saleor/graphql/account/dataloaders.py

```diff
@@ -49,3 +64,64 @@ def batch_load(self, keys):
                 (thumbnail.user_id, thumbnail.size, format)
             ] = thumbnail
         return [thumbnails_by_user_size_and_format_map.get(key) for key in keys]
+
+
+class UserByEmailLoader(DataLoader):
+    context_key = "user_by_email"
+
+    def batch_load(self, keys):
+        user_map = (
+            User.objects.using(self.database_connection_name)
+            .filter(is_active=True)
+            .in_bulk(keys, field_name="email")
+        )
+        return [user_map.get(email) for email in keys]
+
+
+class PermissionByCodenameLoader(DataLoader):
+    context_key = "permission_by_codename"
+
+    def batch_load(self, keys):
+        permission_map = (
+            Permission.objects.filter(codename__in=keys)
+            .prefetch_related("content_type")
+            .in_bulk(field_name="codename")
+        )
+        return [permission_map.get(codename) for codename in keys]
+
+
+def load_user_from_request(request):
```

Comment:
Not sure if this is proper place.  It looks like something more related to auth layer, and processing a JWT token than `dataloaders`. I would say that a better place for this is `saleor/core/auth_backend.py` especially, that it is used only there.

Reply:
Good point, thank you. Fixed.